### DISCUSSION TITLE
how to config vim-table-mode in lua

### DISCUSSION THREAD

**Discussion by qizidog** [LazyVim/LazyVim#1093] [Q&A] [answered]
I am using the [vim-table-mode](https://github.com/dhruvasagar/vim-table-mode) plugin and the official website provides the following configuration code, but I don't know how to configure it using Lua. Can someone help me?
```vimscript
function! s:isAtStartOfLine(mapping)
  let text_before_cursor = getline('.')[0 : col('.')-1]
  let mapping_pattern = '\V' . escape(a:mapping, '\')
  let comment_pattern = '\V' . escape(substitute(&l:commentstring, '%s.*$', '', ''), '\')
  return (text_before_cursor =~? '^' . ('\v(' . comment_pattern . '\v)?') . '\s*\v' . mapping_pattern . '\v$')
endfunction

inoreabbrev <expr> <bar><bar>
          \ <SID>isAtStartOfLine('\|\|') ?
          \ '<c-o>:TableModeEnable<cr><bar><space><bar><left><left>' : '<bar><bar>'
inoreabbrev <expr> __
          \ <SID>isAtStartOfLine('__') ?
          \ '<c-o>:silent! TableModeDisable<cr>' : '__'
```

**Comment by dpetka2001** [comment, marked as answer]
You can still use Vimscript inside Lua with `vim.cmd`. See `:h vim.cmd`, it also has examples. So I think that if you wrap your block of code inside `vim.cmd` it should work.

**Reply by qizidog** [reply]
This is definitely the most effective way! I tried using lua to rewrite this configuration but got the result of recursive keybind 🤣
